### ISSUE TITLE
Launching apps sometimes causes libinput-gestures to stop seeing gestures

### ISSUE THREAD

**Issue by jshoham** [bulletmark/libinput-gestures#220]
ibinput-gestures -l output:
```
libinput-gestures: session ubuntu+x11 on Linux-4.18.0-21-generic-x86_64-with-Ubuntu-18.04-bionic, python 3.6.8, libinput 1.10.4
/usr/bin/libinput-gestures: hash 39c01b89dcf284f8153c0e4205ed7c9b
Gestures configured in ~/.config/libinput-gestures.conf:
swipe up         3 xdotool key super
swipe down       3 xdotool key super
swipe left       3 xdotool key alt+Right
swipe right      3 xdotool key alt+Left
swipe right_up   3 xdotool key control+Tab
swipe left_up    3 xdotool key control+shift+t
swipe right_down 3 xdotool key control+w
swipe left_down  3 xdotool key control+shift+Tab
swipe up         4 _internal ws_up
swipe down       4 _internal ws_down
swipe left       4 nautilus /home/jake
swipe right      4 gnome-terminal --working-directory=/home/jake
swipe right_up   4 gedit
swipe left_up    4 xdotool key control+super+d
swipe right_down 4 xdotool key control+super+d
swipe left_down  4 sensible-browser
pinch out        3 gnome-terminal --working-directory=/home/jake
libinput-gestures: device /dev/input/event10: DLL06E4:01 06CB:7A13 Touchpad

```

Sometimes launching an external application prevents libinput-gestures from seeing future gestures. In my case I was able to reproduce it with both gedit and nautilus.

I have the most recent version of libinput-gestures installed and am not experiencing any other problems. 

When the offending program is terminated libinput-gestures will soon execute all the missing gestures all at once, as if they had been piled up and suddenly got released.

Looking at htop I found that the offending programs were being launched as child processes to python/libinput-gestures whenever this issue has appeared:

![htop screenshot 1](https://i.imgur.com/3huutTg.png)

![htop screenshot 2](https://i.imgur.com/RkmLIj1.png)

When everything is working correctly libinput-gestures has one and only one child process, namely libinput-debug-events. The external programs (gedit/nautilus/etc) are launched as independent processes in this case. The correlation of the above observations with the presence or absence of the issue is 100%, so I am fairly confident this is the cause or very closely related.

My guess is that when the offending programs are launched as a child process libinput-gestures starts looking at their std output for gestures, which yields nothing of use and so no gestures can be executed. Meanwhile the actual output of libinput-debug-events continues to build up as unprocessed output lines. When the offending program is terminated libinput-gestures can finally see the correct std output and proceeds to rapidly execute all the missing gestures.

I am not sure what causes the offending programs to be launched as child processes instead of independent processes, and why it only happens sometimes and not all the time.

**Comment by bulletmark** [comment]
This is by design and is how it has always been. libinput-gestures blocks when running the user command so the traditional advice here for the rare case where people want to run a long running process is to invoke a wrapper script which runs that process in the background.

I presume your PR #221 is related to this issue? Thinking about it I guess there is no good reason to block so I will submit a change to do basically as you suggest in that PR and invoke the user command via `Popen'. See my further comments in that PR.

**Comment by bulletmark** [comment]
Closed by commit  29bbb08.
